pages/studentPortal/labs/multiplication-connect/Index.tsx

```diff
@@ -21,11 +21,62 @@ const createGrid = () => {
   return grid;
 }; 
 
+export const calculateWinner = (block, grid) => {
+  // on block press run calculateWinner algo
+
+  /* might be able to use .filter() to break down the grid into rows, verticals, diagonals
+    alternatively can use this: https://stackoverflow.com/questions/32770321/connect-4-check-for-a-win-algorithm 
+  */
+
+  // refactor the checks
+  // (subtract 3 bc 4 in a row can't be formed afterward)
+        /*   // horizontalCheck 
+      for (int j = 0; j<getHeight()-3 ; j++ ){
+        for (int i = 0; i<getWidth(); i++){
+            if (this.board[i][j] == player && this.board[i][j+1] == player && this.board[i][j+2] == player && this.board[i][j+3] == player)
+                return true;
+        }
+      }
+
+       // verticalCheck
+      for (int i = 0; i<getWidth()-3 ; i++ ){
+        for (int j = 0; j<this.getHeight(); j++){
+            if (this.board[i][j] == player && this.board[i+1][j] == player && this.board[i+2][j] == player && this.board[i+3][j] == player){
+                return true;
+            }           
+        }
+      }
+
+      // descendingDiagonalCheck
+      for (int i=3; i<getWidth(); i++){
+        for (int j=3; j<getHeight(); j++){
+            if (this.board[i][j] == player && this.board[i-1][j-1] == player && this.board[i-2][j-2] == player && this.board[i-3][j-3] == player)
+                return true;
+        }
+      } */
+
+      
+  let rows = [grid.filter((i) => (i.id >= 0 && i.id < 5)), 
+              grid.filter((i) => (i.id >= 5 && i.id < 10)),
+              grid.filter((i) => (i.id >= 10 && i.id < 15)),
+              grid.filter((i) => (i.id >= 15 && i.id < 20)),
+              grid.filter((i) => (i.id >= 20 && i.id < 25)), 
+              grid.filter((i) => (i.id >= 25 && i.id < 30)), 
+              grid.filter((i) => (i.id >= 30 && i.id < 35))];
+  // console.log(rows);
+
+  for(let row of rows){
+    for(let id=0; id<2; id++){
```

Comment:
Looks good to me. I would just call this variable i or index instead of Id. Because these aren't the same the block ids